### DISCUSSION TITLE
Are training data saved in fitted model object?

### DISCUSSION THREAD

**Discussion by hiroki32** [CamDavidsonPilon/lifelines#1538] [Q&A] [answered]
Hi,

I just wanted to ask if training data are saved in the fitted model object when using CoxPHFitter.

The reason why I am asking is because there seems to be index names stored in the model object as shown below.

I am planning to distribute my models to others, but I need to make sure that the measurements in training data are not inside.

### Building the model object
```
from lifelines.datasets import load_regression_dataset
from lifelines import CoxPHFitter
import pickle

# load training data 
df_data = load_regression_dataset()

# change index name to 'Subject_###' 
df_data = df_data.reset_index()
df_data['index'] = 'Subject_' + df_data['index'].astype(str).str.zfill(3)
df_data.index = df_data['index']
df_data = df_data.drop(columns='index')

# fit cox model on training data
cph = CoxPHFitter()
cph.fit(df_data, 'T', event_col='E')
cph.print_summary()

# save model object
with open('fitted_cph_model.pickle', mode='wb') as f:
    pickle.dump(cph, f) 
``` 

### Looking inside the model object
Running `less fitted_cph_model.pickle` on a unix terminal shows index names inside the saved model object.
The index names are "Subject_" followed by 3 digit numbers.
![image](https://github.com/CamDavidsonPilon/lifelines/assets/49674297/df203bc6-30fd-4830-8af1-64363dcca9e3)

**Comment by CamDavidsonPilon** [comment]
Hi @hiroki32, 

No, the raw training data is not part of the model, normally. Pieces of the training data are, like the durations, etc. that may contain the original index. 

You can try adding the training data to the model before picking:
```python
cph.df = df 
with open('fitted_cph_model.pickle', mode='wb') as f:
    pickle.dump(cph, f) 
```

**Reply by hiroki32** [reply]
Hi @CamDavidsonPilon 

Thank you for your reply.

> No, the raw training data is not part of the model, **_normally_**.

I have two questions regarding the above part of your answer.

1. You wrote that training data is _**normally**_ not included in the model. In what cases would the original values of training data be stored in the model object.
2. Would using the scripts I've shared in my original post cause the resulting model object to contain any of the original values of the training data.

I am sorry to ask you again, but I need to be absolutely sure that the original values of the training data are not part of the model object.

**Reply by CamDavidsonPilon** [reply, marked as answer]
I understand. I missed your comment about _not_ wanting to include them. I said `Normally` as you can choose to add it, as in my example. 

The durations, censored values, index names, etc. are included on the model.

**Reply by hiroki32** [reply]
Thank you for your answer.
I now understand that fitting the model won't cause the original values of the training data to be contained within the model.
Many thanks for your generous help.
